descartes_light/include/descartes_light/descartes_light.h

```diff
@@ -36,18 +36,18 @@ class Solver
 public:
   Solver(const std::size_t dof);
 
-  bool build(const std::vector<typename PositionSampler<FloatType>::Ptr>& trajectory,
+  bool build(const std::vector<typename WaypointSampler<FloatType>::Ptr>& trajectory,
              const std::vector<typename EdgeEvaluator<FloatType>::Ptr>& edge_eval,
              int num_threads = getMaxThreads());
```

Comment:
I can imagine that there might be many different ways that constructing a graph could fail, so we should throw an exception with a detailed information message on failure here rather than returning a boolean. Also if we change the waypoint sampler and edge evaluator interfaces to be const, we should use `const` unique pointers instead of non-const shared pointers

```suggestion
  void build(const std::vector<typename WaypointSampler<FloatType>::ConstUPtr>& trajectory,
             const std::vector<typename EdgeEvaluator<FloatType>::ConstUPtr>& edge_eval,
             int num_threads = getMaxThreads());
```

Reply:
I switched to `ConstPtr` versus `ConstUPtr` as I am not sure they need to be unique but they must be thread safe which should be documented. What were your thoughts for going with the unique pointer?

Reply:
Discussed offline and will leverage `ConstPtr` for now.